RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETMemberMetricCalculator.cs

```diff
@@ -121,5 +137,404 @@ private int CountOccurrences(string text, string pattern)
+        }
+
+        private int CountUnaryExpressions(MemberDeclarationSyntax method)
+        {
+            int count = method.DescendantNodes().OfType<PrefixUnaryExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.PreIncrementExpression) ||
+                            n.IsKind(SyntaxKind.PreDecrementExpression) ||
+                            n.IsKind(SyntaxKind.UnaryPlusExpression) ||
+                            n.IsKind(SyntaxKind.UnaryMinusExpression));
+            count += method.DescendantNodes().OfType<PostfixUnaryExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.PostIncrementExpression) ||
+                            n.IsKind(SyntaxKind.PostDecrementExpression));
+            return count;
+        }
+
+        private int CountNumberOfParenthesizedExpressions(MemberDeclarationSyntax method)
+        {
+            return method.DescendantNodes().OfType<ExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.ParenthesizedExpression));
+        }
+
+        private int CountNumberOfLambdaExpressions(MemberDeclarationSyntax method)
+        {
+            return method.DescendantNodes().OfType<LambdaExpressionSyntax>().Count();
+        }
+
+        private int CountMaxNestedBlocks(MemberDeclarationSyntax method)
+        {
+            if (!method.Kind().Equals(SyntaxKind.MethodDeclaration))
+            {
+                return 0;
+            }
+            BaseMethodDeclarationSyntax baseMethod = (BaseMethodDeclarationSyntax)method;
+            var blocks = baseMethod.Body.DescendantNodes().OfType<BlockSyntax>().ToList();
+
+            List<int> blocksAncestors = new List<int>();
+            foreach (var block in blocks)
+            {
+                blocksAncestors.Add(block.Ancestors().Count(a => a.IsKind(SyntaxKind.Block)));
+            }
+
+            if (!blocksAncestors.Any())
+            {
+                return 0;
+            }
+            return blocksAncestors.Max();
+        }
+
+        private int CountNumberOfUniqueWords(MemberDeclarationSyntax method)
+        {
+            if (!method.Kind().Equals(SyntaxKind.MethodDeclaration))
+            {
+                return 0;
+            }
+            BaseMethodDeclarationSyntax baseMethod = (BaseMethodDeclarationSyntax)method;
+            string methodBody = baseMethod.Body.ToString();
+            List<string> lines = GetLinesWithoutComments(methodBody);
+            List<string> words = GetWordsFromLines(lines);
+            words = FilterWords(words);
+            List<string> individualWords = BreakWords(words);
+            individualWords = FindAdditionalWords(individualWords);
+            return CleanWordsFromSymbols(individualWords).Distinct().ToList().Count();
+        }
+
+        private List<string> GetLinesWithoutComments(string methodBody)
+        {
+            List<string> lines = methodBody.Split("\n")
+                .Select(line => line.Trim())
```

Comment:
Once you utilize the comment extracting code I comment on below, I think you won't need this trim.